tests/unit/test_synthDrivers/test_espeak.py

```diff
@@ -0,0 +1,101 @@
+# A part of NonVisual Desktop Access (NVDA)
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+# Copyright (C) 2022 NV Access Limited.
+
+"""Unit tests for the eSpeak synth driver submodule.
+"""
+
+import logging
+import unittest
+import logHandler
+from speech.commands import LangChangeCommand
+from synthDrivers.espeak import SynthDriver
+import nvwave
+
+
+class FakeESpeakSynthDriver:
+	_language = "default"
+	_defaultLangToLocale = {"default": "en-gb"}
+	availableLanguages = {"fr", "fr-fr", "en-gb", "ta-ta"}
+
+
+class TestSynthDriver_Logic(unittest.TestCase):
+	"""Testing of internal logic for the eSpeak driver."""
+	def test_normalizeLangCommand(self):
+		"""Test cases for determining a supported eSpeak language from a LangChangeCommand."""
+		self.assertEqual(
+			LangChangeCommand("en-gb"),
+			SynthDriver._normalizeLangCommand(FakeESpeakSynthDriver, LangChangeCommand(None)),
+			msg="Default language used if language code not provided"
+		)
+		self.assertEqual(
+			LangChangeCommand("fr-fr"),
+			SynthDriver._normalizeLangCommand(FakeESpeakSynthDriver, LangChangeCommand("fr_FR")),
+			msg="Language with locale used when available"
+		)
+		self.assertEqual(
+			LangChangeCommand("en-gb"),
+			SynthDriver._normalizeLangCommand(FakeESpeakSynthDriver, LangChangeCommand("default")),
+			msg="Default eSpeak language mappings used"
+		)
+		self.assertEqual(
+			LangChangeCommand("fr"),
+			SynthDriver._normalizeLangCommand(FakeESpeakSynthDriver, LangChangeCommand("fr_FAKE")),
+			msg="Language without locale used when available"
+		)
+		self.assertEqual(
+			LangChangeCommand("ta-ta"),
+			SynthDriver._normalizeLangCommand(FakeESpeakSynthDriver, LangChangeCommand("ta-gb")),
+			msg="Language with any locale used when available"
+		)
+		with self.assertLogs(logHandler.log, level=logging.DEBUG) as logContext:
+			self.assertEqual(
+				LangChangeCommand(None),
+				SynthDriver._normalizeLangCommand(FakeESpeakSynthDriver, LangChangeCommand("fake")),
+				msg="No matching available language returns None"
+			)
+		self.assertIn(
+			"Unable to find an eSpeak language for 'fake'",
+			logContext.output[0]
+		)
+
+
+class TestSynthDriver_Integration(unittest.TestCase):

```

Comment:
For each language (without locale) set the eSpeak language, and then confirm that a voice with that language has been set